GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,328 @@
+        {
+            // given
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+            var invalidConfiguration = CreateRandomConfiguration(dateTimeOffset: randomDateTimeOffset);
+            invalidConfiguration.Name = GetRandomStringWithLengthOf(451);
+
+            var invalidConfigurationException = 
+                new InvalidConfigurationException(
+                message: "Configuration is invalid, fix the errors and try again.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.Name),
+                values: $"Text exceed max length of {invalidConfiguration.Name.Length - 1} characters");
+
+            var expectedConfigurationValidationException =
+                new ConfigurationValidationException(
+                message: "Configuration validation error occurred, fix the errors and try again.",
+                innerException: invalidConfigurationException);
+
+            this.datetimeBrokerMock.Setup(broker => 
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(randomDateTimeOffset);
+
+            // when
+            ValueTask<Configuration> addConfigurationTask = 
+                this.configurationService.AddConfigurationAsync(invalidConfiguration);
+
+            ConfigurationValidationException actualConfigurationValidationException =
+                await Assert.ThrowsAsync<ConfigurationValidationException>(
+                    addConfigurationTask.AsTask);
+
+            // then
+            actualConfigurationValidationException.Should()
+                .BeEquivalentTo(expectedConfigurationValidationException);
+
+            this.loggingBrokerMock.Verify(broker => 
+                broker.LogErrorAsync(It.Is(SameExceptionAs(expectedConfigurationValidationException))), 
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker => 
+                broker.InsertConfigurationAsync(It.IsAny<Configuration>()), 
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfAuditPropertiesAreNotSameAndLogItAsync()
+        {
+            // given
+            DateTimeOffset randomDateTime = GetRandomDateTimeOffset();
+            DateTimeOffset now = randomDateTime;
+            Configuration randomConfiguration = CreateRandomConfiguration(now);
+            Configuration invalidConfiguration = randomConfiguration;
+            invalidConfiguration.CreatedBy = GetRandomString();
+            invalidConfiguration.UpdatedBy = GetRandomString();
+            invalidConfiguration.CreatedDate = now;
+            invalidConfiguration.UpdatedDate = GetRandomDateTimeOffset();
+
+            InvalidConfigurationException invalidConfigurationException = new InvalidConfigurationException(
```

Comment:
Format the same as others.

Reply:
Fixed. 
![image](https://github.com/user-attachments/assets/35b67e73-019f-49d9-b78c-e3a0e0cea973)